src/Fantomas/Context.fs

```diff
@@ -696,6 +700,13 @@ let private shortExpressionWithFallback (shortExpression: Context -> Context)
 let internal isShortExpression maxWidth (shortExpression: Context -> Context) (fallbackExpression) (ctx: Context) =
     shortExpressionWithFallback shortExpression fallbackExpression maxWidth None ctx
 
+let internal isSmallExpression size (smallExpression: Context -> Context) fallbackExpression (ctx: Context) =
+    match size with
+    | CharacterWidth (_width, maxWidth) -> isShortExpression maxWidth smallExpression fallbackExpression ctx
+    | LogicalSize (size, maxSize) ->
+        let effectiveMaxWidth = if size > maxSize then 0 else Num.max
+        isShortExpression effectiveMaxWidth smallExpression fallbackExpression ctx
```

Comment:
No need to call `isShortExpression` here right? if `size > maxSize` you can just `fallbackExpression ctx` otherwise `smallExpression`.

Reply:
I think it is necessary, because `isShortExpression` calls `shortExpressionWithFallback` which contains the logic to make it multiline if it's not under `MaxLineLength`.

Reply:
If `NumberOfItems` is selected 
```fsharp
    | NumberOfItems (items, maxItems) ->
        if items > maxItems then
            fallbackExpression ctx
        else
            smallExpression ctx
```
suffices.

Reply:
Updated to use `expressionFitsOnRestOfLine` as discussed below